### PR TITLE
feat: login page

### PR DESCRIPTION
**Pull request by raosan**
# Symon Pull Request (PR)

### what issue does this pr address

implementing issue no #177  which adds feature login page

### how did you implement / how did you fix it

1. create rememberMe function to save state in localStorage
2. create 2 API endpoints to check is there already user and create first user if no user in DB
3. wire the API to FE

### how **DID** you test

1. npm start
2. check localhost:4000/login
3. if you already seed the DB, login with the seeded user credential (email:admin@symon.org | password:right password). You can check that the access token is stored on localStorage and you will be redirected to home page.
4. open localhost:4000/login again and you will not redirected to home page because "remember me" check is uncheck. Now repeat above step to login again, and when you visit login page again, you will be auto redirected to home page
5. Now test for case of no pre seed user. Delete the DB, then visit login page again and you will see "crate new user" form. Fill the form and the user is created and you will be automatically logged in.

### any other info/context

On a second thought, the seed user is a user zero which can be treated as "already created user". So the one case to display create user form is when there is no user at all in DB. If the seeded user is not treated as user zero, than we cannot log in with that user in the first time, so there is no point in creating the seed

### CODE REVIEW

**Comment by dennypradipta** [comment]
image

Could you please remove the Remember Me in create form?

File: src/server/services/auth/index.ts [hunk 1]
```diff
@@ -18,11 +18,15 @@
...
 import express from "express";
-import { login, refresh } from "./controller";
+import validate from "../../internal/middleware/validator";
+import { createSchemaValidator } from "../users/validator";
+import { login, refresh, checkHasUser, createFirstUser } from "./controller";
 
 const router = express.Router();
 
+router.get("/v1/auth/check-users", checkHasUser);
 router.post("/v1/auth", login);
```
Comment (nicnocquee) [inline src/server/services/auth/index.ts:28]: better to use validate here too to be consistent.

File: src/client/pages/Login/Login.tsx [hunk 2]
```diff
@@ -33,28 +35,80 @@ export const Login: FC = () => {
...
+    if (remember && !!accessToken) {
+      history.replace("/");
+    } else {
+      setData(prev => ({
+        ...prev,
+        remember,
+      }));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (nicnocquee) [inline src/client/pages/Login/Login.tsx:62]: maybe add the history in the dependency here to remove the eslint-disable

**Review by nicnocquee — CHANGES_REQUESTED** [review]

**Comment by raosan** [comment]
> image
> 
> Could you please remove the Remember Me in create new user form?

fixed. please recheck...

**Review by nicnocquee — APPROVED** [review]

**Review by dennypradipta — APPROVED** [review]

**Pull request merged**